pkg/initializer/initializer.go

```diff
@@ -60,22 +65,18 @@ func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int6
 		}
 		m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
 		isScaleup, err := m.IsClusterScaledUp(ctx, clientSet)
-		if isScaleup && err == nil {
-			retry.OnError(retry.DefaultBackoff, errors.IsErrNotNil, func() error {
-				// Additional safety check before adding a learner
-				if _, err := os.Stat(e.Config.RestoreOptions.Config.DataDir); err == nil {
-					if err := os.RemoveAll(filepath.Join(e.Config.RestoreOptions.Config.DataDir)); err != nil {
-						return fmt.Errorf("failed to remove directory %s with err: %v", e.Config.RestoreOptions.Config.DataDir, err)
-					}
-				} else if !os.IsNotExist(err) {
-					return err
-				}
-				return m.AddMemberAsLearner(ctx)
-			})
-			// return here after adding non-voting member(learner) as no restoration or validation needed
-			return nil
-		} else if err != nil {
+		if err != nil {
```

Comment:
if err is nil then it cannot be guaranteed that it is not a scale-up case. I think this is the limitation with which we have to live with at the moment, right?

Reply:
> if err is nil then it cannot be guaranteed that it is not a scale-up case

yes, as you can check when member is already a part of cluster then it will return false,nil
https://github.com/gardener/etcd-backup-restore/blob/1ecfb1a55d9d1969dff6eb245e9e3a2de390d0bc/pkg/member/member_control.go#L330

Reply:
Yes but it is also possible that there is an error trying to make the call `m.IsMemberInCluster`. So in case of an error one ideally cannot assume that it is not part of the cluster. It only means that its membership is `unknown` as we could not determine it.

Reply:
yes